### PR TITLE
Extract selection option labels for Analytics

### PR DESCRIPTION
**Pull request by Rello**
- when column type selection is used, the API needs to deliver the labels, and not the internal ids
fixes https://github.com/Rello/analytics/issues/385

- Using lib/Analytics (and not lib/Datasource) for better readability
- Adopting from LegacyRow to Row

### CODE REVIEW

**Comment by enjeck** [comment]
@Rello Hi! Are you still working on this?

**Comment by Rello** [comment]
Hi @juliushaertl you think we can merge this?
tests look ok to me if I can judge them correctly?

**Comment by blizzz** [comment]
I am afraid this has potential to be breaking potential existing API use cases. I suppose the IDs of the selection can be fetched already (did not double check). Not that the output is much documented.

There is a comflict btw. If we merge, it's better to have the commits squashed.

**Comment by Rello** [comment]
@blizzz what do you suggest?
The idea is nice dynamic and the same for tables & forms (potentially for others). 
An app can register itself to Analytics.
but all the requirements like tests & co are far more that can be achieved by a hobby project.

I think of skipping this and simply ship the datasources for everything with Analytics itself. Kind of double work because local duplication of the API logic might be required in many places. but not impacting other apps

suggestion?
also applies to 
https://github.com/nextcloud/forms/pull/2195

**Comment by Rello** [comment]
as discussed with @blizzz 
the Analytics data source was decoupled from the API endpoint to not create breaking changes.
Analytics is taking care of any specific logic on its own

File: lib/Analytics/AnalyticsDatasource.php [hunk 1]
```diff
@@ -172,13 +176,69 @@ public function readData($option): array {
...
 
-		return [
-			'header' => $header,
-			'dimensions' => array_slice($header, 0, count($header) - 1),
-			'data' => $data,
-			//'rawdata' => $data,
-			'error' => 0,
-		];
+		return ['header' => $header, 'dimensions' => array_slice($header, 0, count($header) - 1), 'data' => $data, //'rawdata' => $data,
+			'error' => 0,];
```
Comment (blizzz) [inline lib/Analytics/AnalyticsDatasource.php:180]: Nitpick-as-a-Service: indentation looks odd. Would be better in the previous style.

**Review by blizzz — APPROVED** [review]
You'll make me extra happy by squashing the commits =)

P.S.: Let me do that quickly

File: lib/Analytics/AnalyticsDatasource.php (hunk 1 above)
Comment (Rello) [reply lib/Analytics/AnalyticsDatasource.php:180]: let me check which code styler setting messed this up

Comment (blizzz) [reply lib/Analytics/AnalyticsDatasource.php:180]: saw to late. but a follow up pr is welcomed.

**Pull request merged**